### ISSUE TITLE
Fatal javascript OOM in Reached heap limit on render html

### ISSUE THREAD

**Issue by nadnerb** [quarto-dev/quarto-cli#2462]
### Bug description

`quarto render xyz.qmd` will throw  `Fatal javascript OOM in Reached heap limit` when a certain number of figures are added (in our case plotly graphs). In our case it was 26 figures and we get OOM.

```
<--- Last few GCs --->

[2027:0x55bc3354a960]    66765 ms: Mark-sweep 1591.0 (1625.9) -> 1587.9 (1622.4) MB, 7.6 / 0.0 ms  (average mu = 0.989, current mu = 0.989) allocation failure; scavenge might not succeed


<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#

/usr/local/bin/quarto: line 128:  2027 Trace/breakpoint trap   (core dumped) "${QUARTO_DENO}" ${QUARTO_ACTION} ${QUARTO_DENO_OPTIONS} ${QUARTO_DENO_EXTRA_OPTIONS} "${QUARTO_IMPORT_ARGMAP}" "${QUARTO_TARGET}" "$@"
```

This is the config we have:

```
---
execute:
  echo: false
jupyter: python3
format:
  html:
    toc: true
    toc-depth: 4
    toc-location: left
    html-math-method: katex
    font-family: Inter,sans-serif
    smooth-scroll: true
    citations-hover: false
    footnotes-hover: false
    code-fold: true
    code-summary: "Show the code"
    code-overflow: wrap
    theme: cosmo
    fontsize: 1.1em
    linestretch: 1.7
---
```

I removed `self-contained` as discussed here: https://github.com/quarto-dev/quarto-cli/issues/1263 

I appologise I do not have a reproducible example. I have tried different linux distos such as Amazon Linux 2 and Ubuntu 22.04. Any ideas on how I might resolve this? Thank you.

### Checklist

- [ ] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [x] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [x] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [x] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cderv** [comment]
Thanks for the report. 

Can you share you quarto version by running `quarto check` ? You could try latest release of Quarto too, if you don't have it yet. 

> I appologise I do not have a reproducible example.

It would really help if you can share a document with which it happens. Otherwise, it will be hard for us to investigate. 
Do you still have the document that creates the issue for you ? 

If you have it, please can you set `QUARTO_PRINT_STACK=true` before rendering with Quarto
````
export QUARTO_PRINT_STACK=true
quarto render document.qmd
````

https://quarto.org/bug-reports.html#dont-hold-back-tell-us-anything-you-think-might-make-a-difference

---- 

Deno related bug : https://github.com/denoland/deno/issues/13408

**Comment by nadnerb** [comment]
Thank you for your response.

The version I am using is `1.1.245`.

```
[✓] Checking Quarto installation......OK
      Version: 1.1.245
      Path: /var/app/software/quarto-1.1.245/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.6 (Conda)
      Path: /var/app/software/miniconda/envs/mdconverter/bin/python
      Jupyter: 4.11.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

```
# export QUARTO_PRINT_STACK=true
# quarto render ${working_dir}/qc_report.qmd

Executing 'qc_report.ipynb'
  Cell 1/35...Done
  Cell 2/35...Done
  Cell 3/35...Done
  Cell 4/35...Done
  Cell 5/35...Done
  Cell 6/35...Done
  Cell 7/35...Done
  Cell 8/35...Done
  Cell 9/35...Done
  Cell 10/35...Done
  Cell 11/35...Done
  Cell 12/35...Done
  Cell 13/35...Done
  Cell 14/35...Done
  Cell 15/35...Done
  Cell 16/35...Done
  Cell 17/35...Done
  Cell 18/35...Done
  Cell 19/35...Done
  Cell 20/35...Done
  Cell 21/35...Done
  Cell 22/35...Done
  Cell 23/35...Done
  Cell 24/35...Done
  Cell 25/35...Done
  Cell 26/35...Done
  Cell 27/35...Done
  Cell 28/35...Done
  Cell 29/35...Done
  Cell 30/35...Done
  Cell 31/35...Done
  Cell 32/35...Done
  Cell 33/35...Done
  Cell 34/35...Done
  Cell 35/35...Done


<--- Last few GCs --->

[3243:0x55b979e889b0]    66511 ms: Mark-sweep 1069.4 (1103.7) -> 1066.5 (1102.9) MB, 5.8 / 0.0 ms  (+ 0.1 ms in 3 steps since start of marking, biggest step 0.1 ms, walltime since start of marking 250 ms) (average mu = 0.990, current mu = 0.993) finalize [3243:0x55b979e889b0]    67111 ms: Mark-sweep 1591.1 (1626.1) -> 1587.9 (1621.9) MB, 8.4 / 0.0 ms  (average mu = 0.988, current mu = 0.986) allocation failure; scavenge might not succeed


<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#

/usr/bin/quarto: line 128:  3243 Trace/breakpoint trap   "${QUARTO_DENO}" ${QUARTO_ACTION} ${QUARTO_DENO_OPTIONS} ${QUARTO_DENO_EXTRA_OPTIONS} "${QUARTO_IMPORT_ARGMAP}" "${QUARTO_TARGET}" "$@"
```

Thanks. It does look like a memory leak and likely related to the deno bug you linked.

**Comment by infusini** [comment]
Hi, 

I managed to create an example that reproduces the error:

[test.txt](https://github.com/quarto-dev/quarto-cli/files/9607456/test.txt)

`quarto check`

```
[✓] Checking Quarto installation......OK
      Version: 1.1.251
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.6 (Conda)
      Path: /data/miniconda3/envs/mdconverter/bin/python
      Jupyter: 4.11.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.0
      Path: /usr/lib/R
      LibPaths:
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      rmarkdown: (None)

      The rmarkdown package is not available in this R installation.
      Install with install.packages("rmarkdown")
```

```
# export QUARTO_PRINT_STACK=true
# quarto render ${working_dir}/test.qmd

Executing 'test.ipynb'
  Cell 1/1...Done


<--- Last few GCs --->

[366414:0x55e9ce5d0950]    25360 ms: Mark-sweep (reduce) 1438.2 (1440.2) -> 1391.7 (1393.7) MB, 1.6 / 0.0 ms  (+ 2.5 ms in 1 steps since start of marking, biggest step 2.5 ms, walltime since start of marking 33 ms) (average mu = 0.937, current mu = 0.934)[366414:0x55e9ce5d0950]    25393 ms: Mark-sweep (reduce) 1484.6 (1486.6) -> 1438.1 (1440.2) MB, 4.4 / 0.0 ms  (average mu = 0.914, current mu = 0.867) allocation failure; scavenge might not succeed


<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#

/opt/quarto/bin/quarto: line 128: 366414 Trace/breakpoint trap   (core dumped) "${QUARTO_DENO}" ${QUARTO_ACTION} ${QUARTO_DENO_OPTIONS} ${QUARTO_DENO_EXTRA_OPTIONS} "${QUARTO_IMPORT_ARGMAP}" "${QUARTO_TARGET}" "$@"
```

**Comment by cderv** [comment]
Thanks a lot. 

I can reproduce using latest pre-release of Quarto 1.2.133

Same issue 
````
Starting python3 kernel...Done

Executing 'test2.ipynb'
  Cell 1/1...Done


<--- Last few GCs --->

[31548:0000023D254FEB80]    43004 ms: Mark-sweep 1392.6 (1436.0) -> 1346.1 (1389.0) MB, 5.5 / 0.0 ms  (average mu = 0.894, current mu = 0.857) allocation failure; scavenge might not succeed
[31548:0000023D254FEB80]    43049 ms: Mark-sweep 1439.1 (1482.0) -> 1439.1 (1482.0) MB, 7.6 / 0.0 ms  (average mu = 0.863, current mu = 0.831) allocation failure; scavenge might not succeed


<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#
`````

@dragonstyle probably something to look into rather quickly. Unless it is known limitation of Deno somehow... :think

**Comment by dragonstyle** [comment]
@cderv Taking a look now - thx for pinging!

**Comment by cderv** [comment]
Ok great. Tell me if you need me to look into it as I can reproduce!

**Comment by dragonstyle** [comment]
OK I've done a bit of experimentation - the basic issue comes down to the fact that the sample QMD results in a notebook with a number of 50MB cells and a total size of 440+MB. We just currently don't handle this all that well (there are a number of places that we're processing the cells and markdown that we'll have to do some serious work to get out of memory to avoid this). 

I experimented with tinkering with v8 options to increase heap size, but that also didn't remedy it (not surprising since there shouldn't be a cap anyhow).

I'm leaving this open but retargeting for 1.3 as it is a big work item for 1.2.

**Comment by elgabbas** [comment]
I am having the same problem when previewing a quarto website using v1.3.14

**Comment by bomeara** [comment]
I'm still having the same issue on both stable and most recent pull. OS X 12.6, Apple M1 Pro chip, 16 GB RAM. The same file with `rmarkdown::render` runs fine, but with `quarto::quarto_render` (and relevant argument names changed) fails with memory issues. My file loads in a large data.frame of data and renders many images as standalone png files (not embedded in the html, just called as img).

```
<--- Last few GCs --->

[81216:0x150008000]     2085 ms: Mark-sweep 801.5 (854.1) -> 799.3 (842.6) MB, 2.9 / 0.1 ms  (+ 0.0 ms in 1 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 1066 ms) (average mu = 0.997, current mu = 0.998) finalize incre[81216:0x150008000]     2359 ms: Mark-sweep 1583.0 (1622.9) -> 1580.2 (1623.2) MB, 11.3 / 0.0 ms  (average mu = 0.987, current mu = 0.959) allocation failure; scavenge might not succeed


<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#

/usr/local/bin/quarto: line 147: 81216 Trace/BPT trap: 5       "${QUARTO_DENO}" ${QUARTO_ACTION} ${QUARTO_DENO_OPTIONS} ${QUARTO_DENO_EXTRA_OPTIONS} "${QUARTO_IMPORT_ARGMAP}" "${QUARTO_TARGET}" "$@"
```

Happened on both quarto 1.2.269 from https://quarto.org/docs/get-started/ and the development version I installed from a git clone taken today (25 Nov 2022):

```
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      Path: /Users/bomeara/Documents/MyDocuments/GitClones/quarto-cli/package/dist/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.8
      Path: /opt/homebrew/opt/python@3.10/bin/python3.10
      Jupyter: 5.0.0
      Kernels: python3
```

Thanks for all the work on quarto -- I'm eager to start using it, but this issue is blocking my use. It's possible I'm doing something remarkably stupid in my markdown, though.

**Comment by elgabbas** [comment]
I was able to avoid this by removing the `embed-resources: true` option. This has substantially reduced the file size of resulted HTML pages from >250MB per file to just some KBs.

**Comment by bomeara** [comment]
Thanks, but no, still failed with same error. My final rendered html files using rmarkdown are only 43 KB (already using the `    self_contained: false` equivalent to `embed-resources: true`), though some of the objects in memory during rendering are pretty large.

**Comment by m-jenkin** [comment]
I'd also like to thank the Quarto team for all their work! Looking forward for this issue to be resolved. I'm using large datasets and lots of interactive graphs for a report but cannot render it.

**Comment by dragonstyle** [comment]
Transferring to Carlos as he is doing work / bug fixes in this area.

**Comment by cscheid** [comment]
We have a new [troubleshooting page](https://quarto.org/docs/troubleshooting/index.html#out-of-memory-issues) in our docs that includes some workarounds for v8 heap limits. In the future we'll try to be smarter about setting up this value automatically, but in the meantime, maybe folks can experiment with this.

**Comment by cscheid** [comment]
I'm going to move this to an enhancement for 1.4, since we document `--v8-flags=--max-old-space-size=` option in our [troubleshooting page](https://quarto.org/docs/troubleshooting/index.html#out-of-memory-issues).

**Comment by dougwyu** [comment]
I got this error and then got rid of the error when i removed a table that looked like this:

| diagnostic              | value                | should be |
|-------------------------|----------------------|-----------|
| ess_L                   | `r ess_L`            | \>150     |
| ess_U                   | `r ess_U`            | \>150     |
| ess_beta0_psi           | `r ess_beta0_psi`    | \>150     |
| Gelman L_output         | `r L_output`         | \< 1.2    |
| Gelman U_output         | `r U_output`         | \< 1.2    |
| Gelman beta0_psi_output | `r beta0_psi_output` | \< 1.2    |

**Comment by nwvhb** [comment]
According to #4368 , the setting of flags is not working with the release versions.  Is that correct? 

If so, this matches what I see on Windows. I have verified that the environment variable is set, but deno still keeps hitting an OOM.   In this case, I've set self-contained to true.  I don't have any memory issues when this is not set. 

```
<--- Last few GCs --->

[33360:000002054061CC50]    15876 ms: Scavenge 1368.2 (1418.5) -> 1367.9 (1430.3) MB, 9.8 / 0.0 ms  (average mu = 0.209, current mu = 0.165) allocation failure;
[33360:000002054061CC50]    15892 ms: Scavenge 1375.6 (1430.3) -> 1376.3 (1431.0) MB, 13.3 / 0.0 ms  (average mu = 0.209, current mu = 0.165) allocation failure;
[33360:000002054061CC50]    15911 ms: Scavenge 1376.3 (1431.0) -> 1375.6 (1454.3) MB, 19.3 / 0.0 ms  (average mu = 0.209, current mu = 0.165) allocation failure;


<--- JS stacktrace --->


#
# Fatal javascript OOM in Reached heap limit
#
```

**Comment by jonpeake** [comment]
Any update on this? I've had the same issue where the release versions won't recognize the QUARTO_DENO_EXTRA_OPTIONS environment variable, while the GitHub version does. While this is fine for my personal needs, it would be great if we could get a release version that integrates the memory limit, particularly for reproducibility purposes (i.e., I'm developing a quarto project with the intention of eventually being used by others across my organization).

**Comment by cscheid** [comment]
@jonpeake I still can't reproduce this issue where released versions don't recognize `QUARTO_DENO_EXTRA_OPTIONS`. I just tested on 1.3 and 1.4 releases. Can you provide more details about your environment?

**Comment by jonpeake** [comment]
@cscheid Windows 10, RStudio version 2023.03.1, R version 4.2.1. Added the environment variable `QUARTO_DENO_EXTRA_OPTIONS="--v8flags=--max-old-space-size=8192"` as a user variable. I've also tried it with `QUARTO_DENO_EXTRA_OPTIONS="--inspect-brk"` as referenced in #4350, but neither seem to do anything when run with release versions 1.3 and 1.4. The OOM error is what has been causing me issues.

**Comment by cscheid** [comment]
I'm on Mac. It must be that our Windows launcher is not honoring this option. @cderv I'm going to assign this to you; we can talk off-thread about it.

**Comment by sayantikabanik** [comment]
Also running into same issue, currently building on M2 and my quarto version is `1.3x`
Have tried setting the env variable and assigning a higher size which didn't help.

The `.qmd` I am trying to render has multiple plots and the input data size is ~500 MB

**Comment by cderv** [comment]
> It must be that our Windows launcher is not honoring this option.

@cscheid Yes I believe this is the reason. On windows, using a dev version of Quarto installed from git repo should not have the issue because `quarto.cmd` has the env var. 

But for pre-release and release builds, `quarto.exe` is used, the Rust launcher does not take into account this option it seems  from what I can read of https://github.com/quarto-dev/quarto-cli/blob/main/package/launcher/src/main.rs

Is the launcher also used on other platform ? 

@dragonstyle I think this is the same as https://github.com/quarto-dev/quarto-cli/commit/ee0c69d4f4de80b8dd49194e46ac16b1c5d826d0 to solve https://github.com/quarto-dev/quarto-cli/issues/4394 and we should be sure the launcher respect all the same configuration env var that our dev version script. Unless we skip them on purpose.

**Comment by dragonstyle** [comment]
This does indeed require us to properly forward this variable along when set, which we are currently not doing. I will have a look.

**Comment by geoallan** [comment]
I found the problem， because I generated the map by color when rendering the map. As a result, four dynamic interactive maps were generated at the same time, resulting in a huge web page, which was very trying to render.

**Comment by dmontecino** [comment]
I had this problem with a quarto book project. It started when I tried to generate self contain HTML. I deleted the HTML files in the _book folder, previewed render again and it was solved

**Comment by geoallan** [comment]
You should check if a large picture or dynamic picture is embedded in the
file, it may cause this problem

dmontecino ***@***.***> 于2023年7月7日周五 03:16写道：

**Comment by philipgladwin** [comment]
Any update on this as I am in the same boat as others where  I can't render because of this issue.

**Comment by dragonstyle** [comment]
We have fixed the windows launcher to forward extra options to deno, so theoretically, the environment variable described [here](https://github.com/quarto-dev/quarto-cli/issues/2462#issuecomment-1579340609) should be respected on windows (in latest pre-release build of Quarto):

`QUARTO_DENO_EXTRA_OPTIONS="--v8flags=--max-old-space-size=8192"`

Have you given that a try?

**Comment by allixender** [comment]
I had to use a slightly different flag:

```bash
Found argument '--v8flags' which wasn't expected, or isn't valid in this context

        Did you mean '--v8-flags'?

        If you tried to supply `--v8flags` as a value rather than a flag, use `-- --v8flags`
```

- Quarto check

```bash
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12 (Conda)
      Path: /home/geoadmin/micromamba/envs/geopython2023/bin/python
      Jupyter: 5.4.0
      Kernels: python3, pangeobinder2022a

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.1
      Path: /usr/lib/R
      LibPaths:
        - /home/geoadmin/R/x86_64-pc-linux-gnu-library/4.3
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
```

- then running with the option worked:

```bash
QUARTO_DENO_EXTRA_OPTIONS="--v8-flags=--max-old-space-size=8192" quarto render mydoc.Rmd --to html
```

**Comment by cscheid** [comment]
Yes, that was a typo, thanks for correcting it!

**Comment by ercbk** [comment]
Just got this error with Quarto 1.4 
![Screenshot (1800)](https://github.com/quarto-dev/quarto-cli/assets/25374582/5e3f356e-b48b-442d-881a-ac5070d0b082)
![Screenshot (1801)](https://github.com/quarto-dev/quarto-cli/assets/25374582/d94a439f-af56-431d-ae55-a6de18156bd7)
![Screenshot (1802)](https://github.com/quarto-dev/quarto-cli/assets/25374582/e7dc4732-f1a5-4c26-b6e3-130821215856)

**Comment by cscheid** [comment]
@ercbk I can repro your issue locally, thank you for the clean github repo to run. Did you try setting 

QUARTO_DENO_EXTRA_OPTIONS="--v8-flags=--max-old-space-size=8192"

on your GHA configuration? I'm not sure what the physical memory limits on GitHub runners are. I'm trying to figure out the smallest amount of memory you'll need to build your project with v1.4. We definitely want to fix this, but maybe we can get you unstuck right away.

**Comment by ercbk** [comment]
I working on setting that environment variable right now.

**Comment by cscheid** [comment]
Locally, 8192MB is enough, but that might run afoul of GHA. I'm trying smaller sizes now.

**Comment by ercbk** [comment]
Success ✔ 8192MB wasn't a problem.

**Comment by cscheid** [comment]
> Success ✔ 8192MB wasn't a problem for GHA. Thanks for the help.

That's good to know. @dragonstyle I wonder if we should just enable this by default.

**Comment by gl2324** [comment]
For everybody searching for a fix of this problem, setting 

`QUARTO_DENO_EXTRA_OPTIONS="--v8-flags=--max-old-space-size=8192"`

worked for me. Thank you

**Comment by mcanouil** [comment]
Indeed, that is the suggestion made earlier: https://github.com/quarto-dev/quarto-cli/issues/2462#issuecomment-1910469717

**Comment by jefunes** [comment]
I am having the same issue. I tried the following suggested solution "#env: QUARTO_DENO_EXTRA_OPTIONS="--v8-flags=--max-old-space-size=8192" but the problem persists. Window Quarto version: ‘1.2.269’

**Comment by mcanouil** [comment]
@jefunes please upgrade your Quarto version.

**Comment by jefunes** [comment]
Ok. I know it should be a straightforward way to do it, how do I update to a new quarto version?  I did the "check for Package Updates" option but it did not work.

**Comment by mcanouil** [comment]
If you are using the built-in Quarto version inside RStudio, upgrade RStudio.

If you have Quarto in another way, then see https://quarto.org/docs/get-started/

**Comment by jefunes** [comment]
I did update both R and Quarto, now I have a different issue. After typing: quarto render the "qmd file"  the session never ends.

**Comment by jefunes** [comment]
It never r renders

**Comment by mcanouil** [comment]
@jefunes Please open a new GitHub discussion as this is unrelated to the current issue and provide sufficient information for us to be able to help. Unfortunately, "I have a problem" does not help us to help you. See <https://quarto.org/bug-reports.html>.

**Comment by prs513rosewood** [comment]
Sorry to comment on a closed issue, but I'm also encountering this problem. On my machine I had to increase the heap limit past the 8192 default introduced in recent quarto versions. However I also run a `quarto render` on a gitlab.com runner and the job [gets killed](https://gitlab.com/prs513rosewood/projets-numeriques-discrets/-/jobs/8260070201), likely because of a memory overrun. Increasing the heap limit does not help there (this used to build [successfully](https://gitlab.com/prs513rosewood/projets-numeriques-discrets/-/jobs/6751204238) six months ago with quarto 1.3.450).

The repository in question is https://gitlab.com/prs513rosewood/projets-numeriques-discrets, here are the steps to reproduce:

```bash
git clone https://gitlab.com/prs513rosewood/projets-numeriques-discrets.git
cd projets-numeriques-discrets/
python3 -m venv venv && source venv/bin/activate && pip install -r requirements.txt
quarto render book
# I run with the execute flag, but the error is the same
```

Here's the output of `quarto check`:

```
Quarto 1.6.32
[✓] Checking environment information...
      Quarto cache location: /home/frerot/.cache/quarto
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.4.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.46.3: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.6.32
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      Chromium: 869685
      TinyTeX: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /opt/texlive/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.10
      Path: /home/frerot/Documents/repos/test/venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

And here is the failed output (with `QUARTO_PRINT_STACK=true`):

```
[ 1/18] index.qmd
[WARNING] Could not fetch resource ./equations_de_mouvement.html
[ 2/18] equations_de_mouvement.ipynb
[WARNING] Could not fetch resource ./interactions.html
[ 3/18] interactions.ipynb
[WARNING] Could not fetch resource ./lj.html
[ 4/18] lj.ipynb
[WARNING] Could not fetch resource ./sph.html
[ 5/18] sph.ipynb
[WARNING] Could not fetch resource ./prep.html
[ 6/18] prep.ipynb
[WARNING] Could not fetch resource ./balistique.html
[ 7/18] balistique.ipynb
[WARNING] Could not fetch resource ./collisions_entre_particules.html
[ 8/18] collisions_entre_particules.ipynb
[WARNING] Could not fetch resource ./lj_tp.html
[ 9/18] lj_tp.ipynb
[WARNING] Could not fetch resource ./thermostat.html
[10/18] thermostat.ipynb
[WARNING] Could not fetch resource ./truss.html
[11/18] truss.ipynb
[WARNING] Could not fetch resource ./plaque.html
[12/18] plaque.qmd
[WARNING] Could not fetch resource ./fissure.html
[13/18] fissure.qmd
[WARNING] Could not fetch resource ./granulaire.html
[14/18] granulaire.qmd
[WARNING] Could not fetch resource ./surface.html
[15/18] surface.qmd
[WARNING] Could not fetch resource ./mosh_pit.html
[16/18] mosh_pit.qmd
[WARNING] Could not fetch resource ./fluides.html
[17/18] fluides.ipynb
[18/18] report.qmd

WARN: lj.html: Unable to resolve crossref @fig-lj-energie
WARN: lj.html: Unable to resolve crossref @fig-lj-energie
WARN: lj.html: Unable to resolve crossref @fig-lj-force
WARN: lj.html: Unable to resolve crossref @fig-lj-cut
WARN: lj.html: Unable to resolve crossref @fig-lj-cut
WARN: lj_tp.html: Unable to resolve crossref @fig-lattice
WARN: truss.html: Unable to resolve crossref @fig-truss

<--- Last few GCs --->

[11478:0x5557c8a15000]   115406 ms: Mark-Compact 8255.6 (8260.8) -> 8255.6 (8260.0) MB, pooled: 6 MB, 31.23 / 0.00 ms  (average mu = 0.859, current mu = 0.429) allocation failure; scavenge might not succeed


<--- JS stacktrace --->



#
# Fatal JavaScript out of memory: Reached heap limit
#
==== C stack trace ===============================

    /opt/quarto/bin/tools/x86_64/deno(+0x2d8b133) [0x5557c2ca5133]
    /opt/quarto/bin/tools/x86_64/deno(+0x2d8a9fb) [0x5557c2ca49fb]
    /opt/quarto/bin/tools/x86_64/deno(+0x2d85f18) [0x5557c2c9ff18]
    /opt/quarto/bin/tools/x86_64/deno(+0x2ddc1ac) [0x5557c2cf61ac]
    /opt/quarto/bin/tools/x86_64/deno(+0x2f90467) [0x5557c2eaa467]
    /opt/quarto/bin/tools/x86_64/deno(+0x2f8e613) [0x5557c2ea8613]
    /opt/quarto/bin/tools/x86_64/deno(+0x2f85211) [0x5557c2e9f211]
    /opt/quarto/bin/tools/x86_64/deno(+0x2f85ce2) [0x5557c2e9fce2]
    /opt/quarto/bin/tools/x86_64/deno(+0x2f64278) [0x5557c2e7e278]
    /opt/quarto/bin/tools/x86_64/deno(+0x30aae11) [0x5557c2fc4e11]
    /opt/quarto/bin/tools/x86_64/deno(+0x30ae780) [0x5557c2fc8780]
    /opt/quarto/bin/tools/x86_64/deno(+0x30ae7a6) [0x5557c2fc87a6]
    /opt/quarto/bin/tools/x86_64/deno(+0x30ae7a6) [0x5557c2fc87a6]
    /opt/quarto/bin/tools/x86_64/deno(+0x30ae7a6) [0x5557c2fc87a6]
    /opt/quarto/bin/tools/x86_64/deno(+0x30ae7a6) [0x5557c2fc87a6]
    /opt/quarto/bin/tools/x86_64/deno(+0x30aaa2f) [0x5557c2fc4a2f]
    /opt/quarto/bin/tools/x86_64/deno(+0x30aa857) [0x5557c2fc4857]
    /opt/quarto/bin/tools/x86_64/deno(+0x2e222bf) [0x5557c2d3c2bf]
    [0x55576412dc76]
/opt/quarto/bin/quarto: line 192: 11478 Trace/breakpoint trap   (core dumped) "${QUARTO_DENO}" ${QUARTO_ACTION} ${QUARTO_DENO_OPTIONS} ${QUARTO_DENO_EXTRA_OPTIONS} "${QUARTO_IMPORT_MAP_ARG}" "${QUARTO_TARGET}" "$@"
```

And without `QUARTO_PRINT_STACK=true`:
```
[ 1/18] index.qmd
[ 2/18] equations_de_mouvement.ipynb
[ 3/18] interactions.ipynb
[ 4/18] lj.ipynb
[ 5/18] sph.ipynb
[ 6/18] prep.ipynb
[ 7/18] balistique.ipynb
[ 8/18] collisions_entre_particules.ipynb
[ 9/18] lj_tp.ipynb
[10/18] thermostat.ipynb
[WARNING] Could not fetch resource ./truss.html
[11/18] truss.ipynb
[WARNING] Could not fetch resource ./plaque.html
[12/18] plaque.qmd
[WARNING] Could not fetch resource ./fissure.html
[13/18] fissure.qmd
[WARNING] Could not fetch resource ./granulaire.html
[14/18] granulaire.qmd
[WARNING] Could not fetch resource ./surface.html
[15/18] surface.qmd
[WARNING] Could not fetch resource ./mosh_pit.html
[16/18] mosh_pit.qmd
[WARNING] Could not fetch resource ./fluides.html
[17/18] fluides.ipynb
[18/18] report.qmd

ERROR: TypeError: failed to allocate string; buffer exceeds maximum length
Path: /home/frerot/Documents/repos/test/book/_book/lj_tp.html

Stack trace:
Path: /home/frerot/Documents/repos/test/book/_book/lj_tp.html
    at readTextFileSync (ext:deno_fs/30_fs.js:864:10)
    at Object.Deno.readTextFileSync (file:///opt/quarto/bin/quarto.js:5124:25)
    at file:///opt/quarto/bin/quarto.js:98911:31
    at Array.map (<anonymous>)
    at websiteOutputFiles (file:///opt/quarto/bin/quarto.js:98910:8)
    at Object.bookPostRender [as postRender] (file:///opt/quarto/bin/quarto.js:100128:26)
    at renderProject (file:///opt/quarto/bin/quarto.js:85784:28)
    at eventLoopTick (ext:core/01_core.js:175:7)
    at async Command.actionHandler (file:///opt/quarto/bin/quarto.js:90187:32)
    at async Command.execute (file:///opt/quarto/bin/quarto.js:8086:13)
```

Thanks for looking at this, the issue is preventing me from updating my class material.

**Comment by cderv** [comment]
@prs513rosewood I'll move to a new issue to better discussed this as we are preparing for 1.6 and this closed issue was marked 1.5
